### PR TITLE
Fixed event create with blank times

### PR DESCRIPTION
**Pull request by beingmattlevy**
Fix create event issue with empty times in event params
Refactor to put convert for db times functionality into times helper
Added test cases

### CODE REVIEW

File: app/helpers/time_helper.rb [hunk 1]
```diff
@@ -18,5 +18,17 @@ def to_datetime_from_picker(datetime_string)
...
       ::DateTime.strptime("#{datetime_string.upcase} #{Time.current.strftime('%z')}Z", TIME_FORMAT).to_time
     end
+
+    # converts all values in hash to Time where key match '_time'
+    def convert_times_for_db(in_hash)
+      return in_hash if in_hash.nil? || in_hash.blank?
+
+      converted_times = {}
+      in_hash.keys.grep(/_time$/).each do |key|
+        converted_times[key] = TimeHelper.to_datetime_from_picker(in_hash[key]) if in_hash[key].present?
```
Comment (ev1lm0nk3y) [inline app/helpers/time_helper.rb:28]: for my education, you still need to refer to `TimeHelper` here even if it is within the same package/class/file?

Comment (beingmattlevy) [reply app/helpers/time_helper.rb:28]: Its good practice, but not strictly required.

File: spec/helpers/time_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    it 'returns nil with nil' do
+      expect(described_class.convert_times_for_db(nil)).to be_nil
+    end
+
+    it 'returns empty with empty in hash' do
+      expect(described_class.convert_times_for_db({})).to be_empty
+    end
+
+    it 'converts valid time string to datetime' do
+      str = '04/17/2024, 4:20 PM'
```
Comment (kigster) [inline spec/helpers/time_helper_spec.rb:47]: We should verify that it properly converts the time after it's saved into the database.

The test as is would pass even if the time converter was totally wrong.

**Review by kigster — CHANGES_REQUESTED** [review]

File: spec/helpers/time_helper_spec.rb (hunk 2 above)
Comment (beingmattlevy) [reply spec/helpers/time_helper_spec.rb:47]: ::DateTime.strptime does the validation for this call.
We can validate post storage. 
I added a test for an invalid string format

**Pull request closed without merging**